iModelCore/ECDb/ECDb/ECSql/PragmaECSqlPreparedStatement.cpp

```diff
@@ -0,0 +1,1092 @@
+        }
+    }
+    void AppendBisCoreVersion(ECDbCR ecdb, BeJsValue row) {
+        Statement stmt;
+        row.appendValue() = "bis_core_ver";
+        if (auto bisCore = ecdb.Schemas().GetSchema("BisCore")) {
+            row.appendValue() = bisCore->GetSchemaKey().GetVersionString();
+        } else {
+            row.appendValue().SetNull();
+        }
+    }
+    void AppendSQLitePageSize(ECDbCR ecdb, BeJsValue row) {
+        Statement stmt;
+        row.appendValue() = "sqlite_page_size";
+        if (stmt.Prepare(ecdb, "PRAGMA page_size") == BE_SQLITE_OK && stmt.Step() == BE_SQLITE_ROW) {
+            row.appendValue() = stmt.GetValueText(0);
+        } else {
+            row.appendValue().SetNull();
+        }
+    }
+    virtual DbResult Read(PragmaManager::RowSet& rowSet, ECDbCR ecdb, PragmaVal const&)  override {
+        auto result = std::make_unique<StaticPragmaResult>(ecdb);
+        result->AppendProperty("key", PRIMITIVETYPE_String);
+        result->AppendProperty("value", PRIMITIVETYPE_String);
+        result->FreezeSchemaChanges();
+
+        // append rows
+        AppendDbGuid(ecdb, result->AppendRow());
+        AppendProjectId(ecdb, result->AppendRow());
+        AppendParentChangeset(ecdb, result->AppendRow());
+        AppendFileType(ecdb, result->AppendRow());
+        AppendBriefcaseId(ecdb, result->AppendRow());
+        AppendFileName(ecdb, result->AppendRow());
+        AppendConnectionId(ecdb, result->AppendRow());
+        AppendECDbProfileVersion(ecdb, result->AppendRow());
+        AppendECDbCurrentProfileVersion(ecdb, result->AppendRow());
+        AppendECSQLVersion(ecdb, result->AppendRow());
+        AppendCreationDate(ecdb, result->AppendRow());
+        AppendJournalMode(ecdb, result->AppendRow());
+        AppendPageCount(ecdb, result->AppendRow());
+        AppendFreeListCount(ecdb, result->AppendRow());
+        AppendSQLiteVersion(ecdb, result->AppendRow());
+        AppendSQLiteSourceId(ecdb, result->AppendRow());
+        AppendSQLitePageSize(ecdb, result->AppendRow());
+        AppendBisCoreVersion(ecdb, result->AppendRow());
+
+        rowSet = std::move(result);
+        return BE_SQLITE_OK;
+    }
+    virtual DbResult Write(PragmaManager::RowSet& rowSet, ECDbCR ecdb, PragmaVal const&) override {
+        ecdb.GetImpl().Issues().ReportV(IssueSeverity::Error, IssueCategory::BusinessProperties, IssueType::ECSQL, "PRAGMA %s is readonly.", GetName().c_str());
+        rowSet = std::make_unique<StaticPragmaResult>(ecdb);
+        rowSet->FreezeSchemaChanges();
+        return BE_SQLITE_READONLY;
+    }
+    static std::unique_ptr<PragmaManager::Handler> Create () { return std::make_unique<PragmaFileInfo>(); }
+};
+
+//================================================================================
+// @bsiclass PragmaECSchemaVersion
+//================================================================================
+struct PragmaECSchemaVersion : PragmaManager::SchemaHandler {
```

Comment:
Why do we need this?  You can just use ECDbMeta queries.  This syntax is a more compact but we don't need to make it more compact.

Reply:
i have remove the version pragma.